test/e2e/policyrecommendation_test.go

```diff
@@ -0,0 +1,256 @@
+		teardownTest(t, data)
+		deleteRecommendedPolicies(t, data)
+		teardownFlowAggregator(t, data, true)
+	}()
+
+	t.Run("testPolicyRecommendationRun", func(t *testing.T) {
+		testPolicyRecommendationRun(t, data)
+	})
+
+	t.Run("testPolicyRecommendationStatus", func(t *testing.T) {
+		testPolicyRecommendationStatus(t, data)
+	})
+
+	podAIPs, podBIPs, err := createTestPods(data)
+	if err != nil {
+		t.Fatalf("Error when creating test Pods: %v", err)
+	}
+
+	if v4Enabled {
+		srcIP := podAIPs.ipv4.String()
+		dstIP := podBIPs.ipv4.String()
+		testFlow := testFlow{
+			srcIP:      srcIP,
+			dstIP:      dstIP,
+			srcPodName: "perftest-a",
+			dstPodName: "perftest-b",
+		}
+		t.Run("testPolicyRecommendationRetrieve/IPv4", func(t *testing.T) {
+			testPolicyRecommendationRetrieve(t, data, false, testFlow)
+		})
+	}
+	if v6Enabled {
+		srcIP := podAIPs.ipv6.String()
+		dstIP := podBIPs.ipv6.String()
+		testFlow := testFlow{
+			srcIP:      srcIP,
+			dstIP:      dstIP,
+			srcPodName: "perftest-a",
+			dstPodName: "perftest-b",
+		}
+		t.Run("testPolicyRecommendationRetrieve/IPv6", func(t *testing.T) {
+			testPolicyRecommendationRetrieve(t, data, true, testFlow)
+		})
+	}
+}
+
+// Example output: Successfully created policy recommendation job with ID e998433e-accb-4888-9fc8-06563f073e86
+func testPolicyRecommendationRun(t *testing.T, data *TestData) {
+	stdout, err := runJob(t, data)
+	require.NoError(t, err)
+	assert := assert.New(t)
+	assert.Containsf(stdout, "Successfully created policy recommendation job with ID", "stdout: %s", stdout)
+}
+
+// Example output: Status of this policy recommendation job is COMPLETED
+func testPolicyRecommendationStatus(t *testing.T, data *TestData) {
+	stdout, err := runJob(t, data)
+	require.NoError(t, err)
+	stdoutSlice := strings.Split(stdout, " ")
+	jobId := stdoutSlice[len(stdoutSlice)-1]
+	stdout, err = getJobStatus(t, data, jobId)
+	require.NoError(t, err)
+	assert := assert.New(t)
+	assert.Containsf(stdout, "Status of this policy recommendation job is", "stdout: %s", stdout)
+}
+
+// Example output:
+// apiVersion: crd.antrea.io/v1alpha1
+// kind: NetworkPolicy
+// metadata:
+//   name: recommend-allow-anp-fj3hd
+// ...
+func testPolicyRecommendationRetrieve(t *testing.T, data *TestData, isIPv6 bool, testFlow testFlow) {
+	var cmdStr string
+	if !isIPv6 {
+		cmdStr = fmt.Sprintf("iperf3 -c %s", testFlow.dstIP)
+	} else {
+		cmdStr = fmt.Sprintf("iperf3 -6 -c %s", testFlow.dstIP)
+	}
+	stdout, stderr, err := data.RunCommandFromPod(testNamespace, testFlow.srcPodName, "perftool", []string{"bash", "-c", cmdStr})
+	require.NoErrorf(t, err, "Error when running iPerf3 client: %v,\nstdout:%s\nstderr:%s", err, stdout, stderr)
+
+	stdout, err = runJob(t, data)
+	require.NoError(t, err)
+	stdoutSlice := strings.Split(stdout, " ")
+	jobId := stdoutSlice[len(stdoutSlice)-1]
+	err = waitJobComplete(t, data, jobId, jobCompleteTimeout)
+	require.NoErrorf(t, err, "policy recommendation Spark job failed to complete")
```

Comment:
Do we have convention to capital the first letter in this message? As I see you use capital letter in other `require.NoErrorf` messages but not here.

Reply:
changed to capital letter, thanks